### PR TITLE
The fe node get crashed when receive beginRemoteTxn request if it is …

### PR DESCRIPTION
**Pull request by wuleistarrocks**
…not leader.

===========cause=================
When an external table is created and the host property set to a
follower/observer node of the source cluster. If we insert some values
into this external table, the target fe node will receive
beginRemoteTxn rpc, and get crashed because of wal writing
is needed for beginRemoteTxn, which is forbidden in non-leader node.

==========solution===============
For non-leader fe node, it will forward beginRemoteTxn/commitRemoteTxn/
abortRemoteTxn to leader node and return back response from leader.

## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 38 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/master/MasterImpl.java|0|38|00.00%|

**Review by gengjun-git — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/master/MasterImpl.java [hunk 1]
```diff
@@ -1261,6 +1304,23 @@ public TCommitRemoteTxnResponse commitRemoteTxn(TCommitRemoteTxnRequest request)
...
+
+        // if current node is follower, forward it to leader
+        if (!catalog.isMaster()) {
+            try {
+                TNetworkAddress addr = masterAddr();
+                LOG.info("abortRemoteTxn as follower, forward it to master. txn_id: {}, master: {}",
+                         request.getTxn_id(), addr.toString());
+                FrontendService.Client client = ClientPool.frontendPool.borrowObject(addr, 1000);
+                response = client.abortRemoteTxn(request);
+                ClientPool.frontendPool.returnObject(addr, client);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/master/MasterImpl.java:1316]: Better to return object in `finally {}` and also need to invalidate object if the rpc failed.

File: fe/fe-core/src/main/java/com/starrocks/master/MasterImpl.java [hunk 2]
```diff
@@ -1184,9 +1187,33 @@ public TGetTableMetaResponse getTableMeta(TGetTableMetaRequest request) {
...
+                TNetworkAddress addr = masterAddr();
+                LOG.info("beginRemoteTxn as follower, forward it to master. Label: {}, master: {}",
+                         request.getLabel(), addr.toString());
+                FrontendService.Client client = ClientPool.frontendPool.borrowObject(addr, 1000);
+                response = client.beginRemoteTxn(request);
+                ClientPool.frontendPool.returnObject(addr, client);
+                return response;
+            } catch (Exception e) {
+                LOG.warn("create thrift client failed during beginRemoteTxn, label: {}, exception: {}", request.getLabel(), e);
+                throw new TException(e.getMessage());
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/master/MasterImpl.java:1212]: Who will handle the exception？return failed response?

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Pull request merged**